Structure.Rmd

```diff
@@ -0,0 +1,398 @@
+
+The distinction between loading and attaching packages is not important when you're writing scripts, but it's very important when you're writing packages. You'll learn more about the difference and why it's important in [search path](#search-path).
+
+`library()` is not useful when you're developing a package because you have to install the package first. In future chapters you'll learn about `devtools::load_all()` and RStudio's "Build and reload", which allows you to skip install and load a source package directly into memory:
+
+```{r, echo = FALSE}
+knitr::include_graphics("diagrams/loading.png")
+```
+
+## What is a library? {#library}
+
+A library is simply a directory containing installed packages. You can have multiple libraries on your computer. In fact, almost every one has at least two: one for packages you’ve installed, and one for the packages that come with every R installation (like base, stats, etc). Normally, the directories with user-installed packages vary based on the version of R that you’re using. That's why it seems like you lose all of your packages when you reinstall R --- they're still on your hard drive, but R can't find them. 
+
+You can use `.libPaths()` to see which libraries are currently active. Here are mine:
+
+```{r, eval = FALSE}
+.libPaths()
+#> [1] "/Users/hadley/R"                                               
+#> [2] "/Library/Frameworks/R.framework/Versions/3.1/Resources/library"
+lapply(.libPaths(), dir)
+#> [[1]]
+#>   [1] "AnnotationDbi"   "ash"             "assertthat"     
+#>   ...      
+#> [163] "xtable"          "yaml"            "zoo"            
+#> 
+#> [[2]]
+#>  [1] "base"         "boot"         "class"        "cluster"     
+#>  [5] "codetools"    "compiler"     "datasets"     "foreign"     
+#>  [9] "graphics"     "grDevices"    "grid"         "KernSmooth"  
+#> [13] "lattice"      "MASS"         "Matrix"       "methods"     
+#> [17] "mgcv"         "nlme"         "nnet"         "parallel"    
+#> [21] "rpart"        "spatial"      "splines"      "stats"       
+#> [25] "stats4"       "survival"     "tcltk"        "tools"       
+#> [29] "translations" "utils"
+```
+
+The first lib path is for the packages I've installed (I've installed a lot!). The second is for so-called "recommended" packages that come with every installation of R.
+
+When you use `library(pkg)` or `require(pkg)` to load a package, R looks through each path in `.libPaths()` to see if a directory called `pkg` exists. If it doesn't exist, you'll get an error message:
+
+```{r, error = TRUE}
+library(blah)
+# or:
+require(blah)
+```
+
+The main difference between `library()` and `require()` is what happens if a package isn't found. While `library()` throws an error, `require()` prints a warning message and returns FALSE. In practice, this distinction isn't important because when building a package you should __NEVER__ use either inside a package. See [package dependencies](#dependencies) for what you should do instead.
```

Comment:
```suggestion
The main difference between `library()` and `require()` is what happens if a package isn't found. While `library()` throws an error, `require()` prints a warning message and returns `FALSE`. In practice, this distinction isn't important because when building a package you should __NEVER__ use either inside a package. See [package dependencies](#dependencies) for what you should do instead.
```